intermediary_representation.py

```diff
@@ -216,13 +216,33 @@ def __init__(self, scene: bpy.types.Scene, root: Node = Node()):
         self.root = root
         self.scene = scene
 
-    def add_node(self, o: bpy.types.Object, parent: Node = None) -> Node:
+    def add_node(self, o: bpy.types.Object = None, parent: Node = None) -> Node:
         log.debug(f"Adding Blender Object: {o} as a node for root {self.root} and {parent} as parent")
 
-        node = None
+        node = self._translate(o) if o else Node('Placeholder node')
+
+        if self.root is None:
+            self.root = node
+        else:
+            node_parent = parent if parent else self._resolve_parenting(o)
+            node_parent.add_child(node)
+
+        return node
+
+    def _translate(self, o: bpy.types.Object) -> Node:
+        """Translates a Blender object into an IR node / node hierarchy.
+
+        Arguments:
+            o {bpy.types.Object} -- The object to be translated
+
+        Returns:
+            Node -- The node / node hierarchy
+        """
 
         # See https://docs.blender.org/manual/en/dev/editors/3dview/object/types.html
         # See also https://docs.blender.org/manual/en/dev/editors/3dview/object/index.html
+        node = None
```

Comment:
Assigning a default value takes care of the case if o.type is neither MESH, nor CAMERA.

Personally I prefer assigning the value in an else clause, because it forces us to to consider what it means that o.type is neither MESH nor CAMERA.  Is it an unexpected error - should it be reported to developer, or is it a recoverable error (do something) and/or should it perhaps be logged.

A missing "else" after if/elseif is usually a warning sign, similar to a missing "default" in a switch.   Deal with this case somehow - by some error recovery step, or just reporting it, or if truly nothing should be done, explain that in a comment.

Reply:
Oh sorry, I had not expanded all the code below it...  Anyway, you can consider the principle.

Reply:
I think I have covered all light types, so no.

But (1): I might be wrong, and (2): They might add more light types in the future, in which case yes this will silently fail.

Maybe we can standardize a "UnresolvedNode" type? That would push the error elsewhere. The question then becomes what to do with this "UnresolvedNode" when finally translating to RAMSES.

Reply:
By the way, previously I was raising an exception for nodes I could not translate yet but this was bad for development since I'd have to comment it out until the exporter had support for most Blender nodes. And still, I think breaking for something we cannot translate is sort of unnecessarily harsh on the end user, maybe @violinyanev can comment on a better 'default' behavior?

Reply:
>But (1): I might be wrong, and (2): They might add more light types in the future, in which case yes this will silently fail.

Yes, exactly, so you see this is about defensive programming even if it seems that the logic is correct or everything is covered.  It is more a principle.  There is something suspicious about a missing "else".  If we are writing a program that is using asserts to alert about unexpected behaviors (which you have done in other places), then there should be something similar in that else clause, so that it does not continue unnoticed.

(In certain cases it could be correct behavior to continue without any action in "else" (or in "default" of a switch/case) but in that case that could also be noted with a comment)

Reply:
> And still, I think breaking for something we cannot translate is sort of unnecessarily harsh on the end user

For sure, developers and users have different needs.  Sometimes as a developer you are also not happy with the program ending with a fatal error, but sometimes you are.  I think we may have a larger discussion about the various types of error handling strategies.  Maybe best done in a meeting instead of here?